plugins/pluginmanagers/lazy.nix

```diff
@@ -0,0 +1,216 @@
+
+              init =
+                helpers.mkNullOrOption (either bool helpers.nixvimTypes.rawLua)
+                "init functions are always executed during startup";
+
+              config =
+                helpers.mkNullOrOption (either bool helpers.nixvimTypes.rawLua)
+                "config is executed when the plugin loads. The default implementation will automatically run require(MAIN).setup(opts). Lazy uses several heuristics to determine the plugin's MAIN module automatically based on the plugin's name. See also opts. To use the default implementation without opts set config to true.";
+
+              main = helpers.mkNullOrOption str "You can specify the main module to use for config() and opts(), in case it can not be determined automatically. See config()";
+
+              build =
+                helpers.mkNullOrOption
+                (oneOf [
+                  (either str (listOf str))
+                  helpers.nixvimTypes.rawLua
+                ])
+                "build is executed when a plugin is installed or updated. Before running build, a plugin is first loaded. If it's a string it will be ran as a shell command. When prefixed with : it is a Neovim command. You can also specify a list to executed multiple build commands. Some plugins provide their own build.lua which is automatically used by lazy. So no need to specify a build step for those plugins.";
+
+              submodules = helpers.defaultNullOpts.mkBool true "When false, git submodules will not be fetched. Defaults to true";
+
+              event =
+                helpers.mkNullOrOption
+                (oneOf [
+                  (either str (listOf str))
+                  helpers.nixvimTypes.rawLua
+                ])
+                "Lazy-load on event. Events can be specified as BufEnter or with a pattern like BufEnter *.lua";
+
+              cmd =
+                helpers.mkNullOrOption
+                (oneOf [
+                  (either str (listOf str))
+                  helpers.nixvimTypes.rawLua
+                ])
+                "Lazy-load on command";
+
+              ft =
+                helpers.mkNullOrOption
+                (oneOf [
+                  (either str (listOf str))
+                  helpers.nixvimTypes.rawLua
+                ])
+                "Lazy-load on filetype";
+
+              keys =
+                helpers.mkNullOrOption
+                (oneOf [
+                  (either str (listOf str))
+                  helpers.nixvimTypes.rawLua
+                ])
+                "Lazy-load on key mapping";
+
+              module = helpers.defaultNullOpts.mkBool false "Do not automatically load this Lua module when it's required somewhere";
+
+              priority = helpers.defaultNullOpts.mkNum null "Only useful for start plugins (lazy=false) to force loading certain plugins first. Default priority is 50. It's recommended to set this to a high number for colorschemes.";
```

Comment:
Don't use a `null` as a default with `defaultNullOpts`, pleas use the `mkNullOrOption` directly